packages/lib/src/signin/Connect.ts

```diff
@@ -25,19 +22,19 @@ export async function connectAccount(
     try {
         const account =
             preSetAccount ?? (await requestAccounts(connection.provider));
-        const profile = await getProfileRegistryEntry(
+        const profile = await getUserProfile(
             connection,
             account,
             connection.defaultServiceUrl + '/profile/' + account,
         );
-        if (profile && !checkProfileRegistryEntry(profile, account)) {
+        if (profile && !checkUserProfile(profile, account)) {
             throw Error('Profile signature is invalid');
         }
+        //Todo is this the right way to do it
+
         return {
             account,
-            existingAccount: profile?.profileRegistryEntry.publicKeys
-                ? true
-                : false,
+            existingAccount: profile !== undefined,
```

Comment:
Is this the proper way to check if an account already exists? Went with the check if the profile is undefined because according to the interface it's not possible that a key is undefined.

Reply:
Let us discuss this on a call